### PR TITLE
Leak fixes

### PR DESCRIPTION
**Pull request by jeffreylovitz**
A few memory leak fixes in Redis key handles and procedure read-only checks.

### CODE REVIEW

File: src/ast/ast.c [hunk 1]
```diff
@@ -108,8 +108,7 @@ static bool _AST_ReadOnly(const cypher_astnode_t *root) {
 	// In case of procedure call which modifies the graph/indices.
 	if(type == CYPHER_AST_CALL) {
 		const char *proc_name = cypher_ast_proc_name_get_value(cypher_ast_call_get_proc_name(root));
-		ProcedureCtx *proc = Proc_Get(proc_name);
-		return proc->readOnly;
```
Comment (swilly22) [inline src/ast/ast.c]: Good catch

File: src/procedures/procedure.c [hunk 2]
```diff
@@ -118,6 +118,18 @@ bool Procedure_ContainsOutput(const ProcedureCtx *proc, const char *output) {
 	return false;
 }
 
+bool Proc_ReadOnly(const char *proc_name) {
+	assert(__procedures);
+	ProcGenerator gen = raxFind(__procedures, (unsigned char *)proc_name, strlen(proc_name));
+	if(gen == raxNotFound) return false; // Invalid procedure specified, handled elsewhere.
+	/* TODO It would be preferable to be able to determine whether a procedure is read-only
+	 * without creating its entire context; this is wasteful. */
```
Comment (swilly22) [inline src/procedures/procedure.c]: Agree

File: src/procedures/procedure.h [hunk 3]
```diff
@@ -35,11 +35,15 @@ uint Procedure_Argc(const ProcedureCtx *proc);
 /* Return number of outputs yield by procedure. */
 uint Procedure_OutputCount(const ProcedureCtx *proc);
 
-/* Retrieves procedure ith output */
+/* Retrieves procedure with output */
```
Comment (swilly22) [inline src/procedures/procedure.h]: `ith` meaning output at potion `i`.

File: src/procedures/procedure.h [hunk 4]
```diff
@@ -35,11 +35,15 @@ uint Procedure_Argc(const ProcedureCtx *proc);
...
 const char *Procedure_GetOutput(const ProcedureCtx *proc, uint output_idx);
 
 /* Returns true if given output can be yield by procedure */
 bool Procedure_ContainsOutput(const ProcedureCtx *proc, const char *output);
 
+/* Returns true if procedure is read-only. */
+bool Proc_ReadOnly(const char *proc_name);
```
Comment (swilly22) [inline src/procedures/procedure.h]: Nice addition.

File: src/procedures/procedure.h (hunk 3 above)
Comment (jeffreylovitz) [reply src/procedures/procedure.h]: Oh, my mistake!

**Pull request merged**